packages/tests/webcomponentsjs_/event-listener-options.html

```diff
@@ -0,0 +1,371 @@
+          });
+
+          test('same listener with different `once` and `capture` options', function () {
+            // same listener adds 2x with change of capture state
+            el1.addEventListener('click', callback, {once: true});
+            el1.addEventListener('click', callback, {
+              capture: true,
+              once: true,
+            });
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 2, 'called 2x');
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 2, 'calls not increased');
+            // can capture and non-capture can have different once options
+            el1.addEventListener('click', callback, {once: true});
+            el1.addEventListener('click', callback, {
+              capture: true,
+            });
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 4, 'called 2x again');
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 5, 'called for capture only');
+            el1.removeEventListener('click', callback, {
+              capture: true,
+            });
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 5, 'not called after removed');
+            // can remove captured event early.
+            el1.addEventListener('click', callback, {once: true});
+            el1.addEventListener('click', callback, true);
+            el1.removeEventListener('click', callback, true);
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 6, 'called 1x after early remove');
+            // can remove non-captured event early.
+            el1.addEventListener('click', callback, {once: true});
+            el1.addEventListener('click', callback, true);
+            el1.removeEventListener('click', callback);
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 7, 'called 1x after early remove');
+          });
+
+          test('`once` listener called with correct this', function () {
+            let pThisRef, pCurrentTarget;
+            const pListener = function (e) {
+              pThisRef = this;
+              pCurrentTarget = e.currentTarget;
+            };
+            let cThisRef, cCurrentTarget;
+            const cListener = function (e) {
+              cThisRef = this;
+              cCurrentTarget = e.currentTarget;
+            };
+            parent.addEventListener('test', pListener, {once: true});
+            child.addEventListener('test', cListener, {once: true});
+            gchild.dispatchEvent(new Event('test', {bubbles: true}));
+            assert.ok(pThisRef);
+            assert.equal(pThisRef, pCurrentTarget);
+            assert.ok(cThisRef);
+            assert.equal(cThisRef, cCurrentTarget);
+          });
```

Comment:
Could you add one for the `handleEvent` case also?

Reply:
Added and this revealed an issue. I can't find it in the spec, but via empirical testing, `this` is the listener object itself when listening with an object with `handleEvent`.

Reply:
Ok, if this is what browsers do then that sounds good.

Reply:
I misread the spec. From [call a user object's operation](https://heycam.github.io/webidl/#call-a-user-objects-operation), step 10.5:

> Set *thisArg* to *O* (overriding the provided value).

Where *O* is the listener object at that point.